lib/utilities/pristine.js

```diff
@@ -35,111 +55,117 @@ function cloneApp(appName, options) {
-    return installPristineApp(appName, options)
-      .then(function() {
-        copyUnderTestApp(pristineAppPath, underTestAppPath);
-      })
-      .then(function() {
-        chdir(previousCwd);
-        return underTestAppPath;
-      });
+    appInstallation = installPristineApp(appName, options);
+  } else {
+    appInstallation = Promise.resolve();
   }
 
-  copyUnderTestApp(pristineAppPath, underTestAppPath);
-  chdir(previousCwd);
-
-  return Promise.resolve(underTestAppPath);
+  return appInstallation.then(() => {
+    copyUnderTestApp(pristineAppPath, underTestAppPath);
+    chdir(previousCwd);
+    return underTestAppPath;
+  });
 }
 
 module.exports = {
-  cloneApp: cloneApp
+  createApp
 };
 
 function installPristineApp(appName, options) {
-  var hasNodeModules = hasPristineNodeModules();
-  var hasBowerComponents = hasPristineBowerComponents();
-  var extraOptions = [];
-  var emberVersion = options.emberVersion || 'canary';
-  var emberDataVersion = options.emberDataVersion || 'emberjs/data#master';
-
-  // First, determine if we can skip installing npm packages
-  // or Bower components if we have a pristine set of dependencies
-  // already.
-
-  // Fresh install
-  if (!hasNodeModules && !hasBowerComponents) {
-    debug("no node_modules or bower_components");
-  // bower_components but no node_modules
-  } else if (!hasNodeModules && hasBowerComponents) {
-    debug("no node_modules but existing bower_components");
-    extraOptions = ['--skip-bower'];
-  // node_modules but no bower_components
-  } else if (hasNodeModules && !hasBowerComponents) {
-    debug("no bower_components but existing node_modules");
-    extraOptions = ['--skip-npm'];
-  // Everything is already there
-  } else {
-    debug("existing node_modules and bower_components");
-    extraOptions = ['--skip-npm', '--skip-bower'];
-  }
+  let hasNodeModules = hasPristineNodeModules();
+  let hasBowerComponents = hasPristineBowerComponents();
 
   chdir(temp.pristinePath);
 
-  var args = extraOptions.concat(runCommandOptions);
-
-  var promise = runNew(appName, args)
+  // Install a vanilla app and cd into it
+  let args = generateArgsForEmberNew(hasNodeModules, hasBowerComponents);
+  let promise = runNew(appName, args)
     .catch(handleResult)
-    .then(function() {
-      chdir(path.join(temp.pristinePath, appName));
-    })
-    .then(addEmberDataToDependencies(appName, emberDataVersion))
-    .then(removeEmberSourceFromDependencies(appName, emberVersion));
-
-  // If we installed a fresh node_modules or bower_components directory,
-  // grab those as pristine copies we can use in future runs.
+    .then(() => chdir(path.join(temp.pristinePath, appName)));
+
+  let setupOptions = {
+    appName,
+    hasNodeModules,
+    hasBowerComponents,
+    emberVersion: options.emberVersion || 'canary',
+    emberDataVersion: options.emberDataVersion || 'emberjs/data#master'
+  };
+
+  promise = chainNodeModulesSetup(promise, setupOptions);
```

Comment:
Largely bikeshedding, but would something like this be more ideomatic?
```js
promise = promise.then(() => chainNodeModulesSetup(setupOptions));
```

So the `chain*` functions just return a promise, and don't have to care about chaining.